packages/app/app/containers/ArtistViewContainer/ArtistViewContainer.test.tsx

```diff
@@ -0,0 +1,167 @@
+import { render, waitFor } from '@testing-library/react';
+import React from 'react';
+import { createMemoryHistory } from 'history';
+
+import { AnyProps, configureMockStore, setupI18Next, TestRouterProvider, TestStoreProvider } from '../../../test/testUtils';
+import MainContentContainer from '../MainContentContainer';
+import { buildStoreState } from '../../../test/storeBuilders';
+
+describe('Artist view container', () => {
+  beforeAll(() => {
+    setupI18Next();
+  });
+
+  beforeEach(() => {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    const { store } = require('@nuclear/core');
+    store.clear();
+  });
+
+  it('should display an artist', () => {
+    const { component } = mountComponent();
+    expect(component.asFragment()).toMatchSnapshot();
+  });
+
+  it('should show that an artist is still loading', () => {
+    const { component } = mountComponent(
+      buildStoreState()
+        .withPlugins()
+        .withArtistDetails({ ['test-artist-id']: { loading: true } })
+        .build()
+    );
+
+    expect(component.asFragment()).toMatchSnapshot();
+  });
+
+  it('should search for tags after clicking his tag', async () => {
+    const { component, history } = mountComponent();
+    expect(history.location.pathname).toBe('/artist/test-artist-id');
+    await waitFor(() => component.getByText(/#tag1/i).click());
+    expect(history.location.pathname).toBe('/tag/tag1');
+  });
+
+  it('should search for similar artist after clicking his name', async () => {
+    const { component, history, store } = mountComponent();
+    expect(history.location.pathname).toBe('/artist/test-artist-id');
+    await waitFor(() => component.getByText(/artist-similar-1/i).click());
+    const state = store.getState();
```

Comment:
Maybe also check that we went to `artist-similar-1`'s page by checking the current path in history?